### DISCUSSION TITLE
v7 `watch` type in typescript?

### DISCUSSION THREAD

**Discussion by baadillahnabil** [react-hook-form/react-hook-form#7320] [Q&A]
Hi,

I am confused about how to declare a type for `watch` as a prop. Here is an example:

```
// pages/FormContainer.tsx

import { useForm, useWatch } from "react-hook-form";
import FormView from "./FormView";

interface FormInputs {
  firstName: string;
  lastName: string;
  age: number;
}

const FormContainer = () => {
  const { control, register } = useForm<FormInputs>({
    defaultValues: {
      firstName: "Jon",
      lastName: "Doe",
      age: 24
    }
  });
  const formData = useWatch({ control });

  return <FormView register={register} formData={formData} />;
};

export default FormContainer;
```

```
// pages/FormView.tsx

import { UseFormReturn } from "react-hook-form";

interface Props {
  register: UseFormReturn["register"];
  formData: UseFormReturn["watch"];
}

const FormView = ({ formData }: Props) => {
  return (
    <div>
      <h1>
        My Name: {formData.firstName} {formData.lastName}
      </h1>
      <h2>I am {formData.age} years old</h2>
    </div>
  );
};

export default FormView;
```

The thing is typescript shows me an error on this `formData={formData}` prop

Here I provided a [sandbox](https://codesandbox.io/s/fast-snowflake-3yk8z?file=/src/App.tsx) to be clear on what I mean

**Comment by bluebill1049** [comment]
`UseFormReturn["watch"]` is a function type not return type of your formData

**Reply by baadillahnabil** [reply]
I tried using `ReturnType` like this `formData: ReturnType<UseFormReturn["watch"]>;` but it gave me a new error saying `Property 'firstName' does not exist on type 'Subscription'.ts(2339)`

Do I have to use my `FormInput` interface as a type? I mean that'll work but I just wanna make sure that there's no type coming from rhf that can handle this, or am I missing something?

**Reply by felixschorer** [reply]
`watch` has a few overloads. `ReturnType` won't work in that case. It just picks the first overload afaik. Also, `UseFormReturn` is generic on the type of the form fields. If you don't supply that type it basically defaults to `any`.

Can't you just use `FormInputs` as the type of the prop?
